### PR TITLE
Corrected bcdiv and bcmul return types for PHP 8.

### PR DESCRIPTION
**Pull request by khkramer**

### CODE REVIEW

File: tests/PHPStan/Analyser/NodeScopeResolverTest.php [hunk 1]
```diff
@@ -122,7 +122,11 @@ public function dataFileAsserts(): iterable
 		yield from $this->gatherAssertTypes(__DIR__ . '/../Rules/Comparison/data/bug-2550.php');
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-2899.php');
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/preg_split.php');
-		yield from $this->gatherAssertTypes(__DIR__ . '/data/bcmath-dynamic-return.php');
+
+		if (PHP_VERSION_ID < 80000) {
+			yield from $this->gatherAssertTypes(__DIR__ . '/data/bcmath-dynamic-return.php');
```
Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/NodeScopeResolverTest.php:127]: What happens on PHP 8 also needs to be tested.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Fixes https://github.com/phpstan/phpstan/issues/5134

**Comment by staabm** [comment]
@khkramer any plans to finalize this PR?

**Comment by dmitryuk** [comment]
@ondrejmirtes Why PR was closed? The bug is still present

**Pull request closed without merging**